src/sqlancer/postgres/PostgresProvider.java

```diff
@@ -218,24 +220,54 @@ protected TestOracle getTestOracle(PostgresGlobalState globalState) throws SQLEx
 
     @Override
     public Connection createDatabase(PostgresGlobalState globalState) throws SQLException {
-        String url = "jdbc:postgresql://localhost:5432/test";
+        String username = globalState.getOptions().getUserName();
+        String password = globalState.getOptions().getPassword();
+        String entryPath = "/test";
+        String entryURL = globalState.getDmbsSpecificOptions().connectionURL;
+        // trim URL to exclude "jdbc:"
+        if (entryURL.startsWith("jdbc:")) {
+            entryURL = entryURL.substring(5);
+        }
+        try {
+            URI uri = new URI(entryURL);
+            String userInfoURI = uri.getUserInfo();
+            String pathURI = uri.getPath();
+            if (userInfoURI != null) {
+                // username and password specified in URL take precedence
+                String[] userInfo = userInfoURI.split(":", 2);
+                username = userInfo[0];
+                password = userInfo[1];
```

Comment:
What happens when only the username is provided and no password?
I think we will get an out of bound exception on the last line here.

connection url to test: `postgres://foo@localhost:5432/test`

Reply:
I added a check to see if a : separator exists and parse it accordingly. DriverManager.getConnection() seems to safely accept null as a valid password, so I set the password to null if there is no password provided